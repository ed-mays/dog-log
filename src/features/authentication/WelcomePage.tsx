import React from 'react';
import { useTranslation } from 'react-i18next';
import LoginButton from '@components/common/Auth/LoginButton';
import { useAuthUser } from '@store/auth.store';
import { Navigate, useLocation } from 'react-router-dom';
import { useFeatureFlag } from '@featureFlags/useFeatureFlag';

const WelcomePage: React.FC = () => {
  const { t } = useTranslation('common');
  const user = useAuthUser();
  const location = useLocation();
  const authEnabled = useFeatureFlag('authEnabled');

  if (user) {
    const from = (location.state as { from?: Location } | undefined)?.from;
    let to = '/pets';
    if (from && typeof from === 'object' && 'pathname' in from) {
      to = (from as Location).pathname;
    }
    return <Navigate to={to} replace />;
  }

  return (
    <main aria-labelledby="welcome-heading">
      <h1 id="welcome-heading">{t('welcomeHeader', 'Welcome')}</h1>
      {!authEnabled ? (
        <p>
          {t('authDisabledMessage', 'Authentication is currently disabled.')}
        </p>
      ) : (
        <>
          <p>{t('welcomeSubtitle', 'Sign in to continue')}</p>
          <LoginButton />
        </>
      )}
    </main>
  );
};

export default WelcomePage;
