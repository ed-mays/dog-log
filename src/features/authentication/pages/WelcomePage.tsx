import { useTranslation } from 'react-i18next';
import LoginButton from '@features/authentication/components/LoginButton';
import { useEffect, useState } from 'react';
import { loadNamespace } from '@i18n';

export function WelcomePage() {
  const [nsReady, setNsReady] = useState(false);

  useEffect(() => {
    let mounted = true;
    Promise.all([loadNamespace('common')]).then(() => {
      if (mounted) setNsReady(true);
    });
    return () => {
      mounted = false;
    };
  }, []);

  const { t } = useTranslation();

  if (!nsReady) return null;
  return (
    <div className="@container">
      <div className="flex flex-col p4 justify-center">
        <h1 className="text-6xl font-bold pb-10">
          {t('welcomePage.welcomeHeader', 'Welcome to Dog Log!')}
        </h1>
        <p className="text-3xl pb-5">
          {t('welcomePage.welcomeSubtitle', 'Please sign in to continue.')}
        </p>
        <LoginButton className="bg-blue-500 hover:border-blue-950 hover: border-2 text-white p-6 rounded shadow" />
      </div>
    </div>
  );
}
