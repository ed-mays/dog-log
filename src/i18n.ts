import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  lng: import.meta.env.VITE_DEFAULT_LOCALE,
  fallbackLng: 'en',
  ns: ['common'],
  defaultNS: 'common',
  interpolation: { escapeValue: false },
});

// Helper to lazy-load a namespace at runtime
export async function loadNamespace(ns: string, lng: string = i18n.language) {
  // Dynamically import the JSON for the requested namespace and language
  const mod = await import(`./locales/${lng}/${ns}.json`);
  i18n.addResourceBundle(lng, ns, mod.default ?? mod, true, true);
  return ns;
}

export default i18n;
