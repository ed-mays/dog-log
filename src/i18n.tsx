import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import homeEN from './locales/en/home.json';
import commonEN from './locales/en/common.json';
import petListEN from './locales/en/petList.json';
import petPropertiesEN from './locales/en/petProperties.json';

import homeES from './locales/es/home.json';
import commonES from './locales/es/common.json';
import petListES from './locales/es/petList.json';
import petPropertiesES from './locales/es/petProperties.json';

i18n.use(initReactI18next).init({
  resources: {
    en: {
      common: commonEN,
      home: homeEN,
      petList: petListEN,
      petProperties: petPropertiesEN,
    },
    es: {
      common: commonES,
      home: homeES,
      petList: petListES,
      petProperties: petPropertiesES,
    },
  },
  lng: import.meta.env.VITE_DEFAULT_LOCALE,
  fallbackLng: 'en',
  ns: ['common', 'home', 'petList'],
  defaultNS: 'common',
  interpolation: { escapeValue: false },
});
export default i18n;
