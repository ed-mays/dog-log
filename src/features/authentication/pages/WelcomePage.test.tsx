import { render, screen, waitFor } from '@testing-library/react';
import { WelcomePage } from './WelcomePage';
import { Suspense } from 'react';
import i18n from 'i18next';

/*
vi.mock(import('react-i18next'), async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    useTranslation: () => ({
      t: (key: string, fallback: string) => fallback,
    }),
  };
});
*/

describe('WelcomePage', () => {
  it('renders the welcome message and login button', async () => {
    render(<WelcomePage />);

    expect(await screen.findByText('Welcome to Dog Log!')).toBeInTheDocument();
    expect(
      await screen.findByText('Please sign in to continue.')
    ).toBeInTheDocument();
    expect(await screen.findByTestId('login-button')).toBeInTheDocument();
  });

  it('renders the expected content in English', async () => {
    const { asFragment } = render(
      <Suspense fallback={<div />}>
        <WelcomePage />
      </Suspense>
    );
    await waitFor(() => screen.findByTestId('login-button'));
    await waitFor(() => screen.findByText('Please sign in to continue.'));
    await waitFor(() => screen.findByText('Welcome to Dog Log!'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the expected content in Spanish', async () => {
    await i18n.changeLanguage('es');
    const { asFragment } = render(
      <Suspense fallback={<div />}>
        <WelcomePage />
      </Suspense>
    );
    await waitFor(() => screen.findByText('Inicia sesiÃ³n para continuar.'));
    await waitFor(() => screen.findByText('Bienvenido a Dog Log!'));
    expect(asFragment()).toMatchSnapshot();
  });
});
