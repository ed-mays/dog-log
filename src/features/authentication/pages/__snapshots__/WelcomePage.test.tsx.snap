// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomePage > renders the expected content in English 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-omazcq-MuiStack-root"
  >
    <h1>
      Welcome to Dog Log!
    </h1>
    <p>
      Please sign in to continue.
    </p>
    <button
      aria-label="Continue with Google"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
      data-testid="login-button"
      tabindex="0"
      type="button"
    >
      Continue with Google
    </button>
  </div>
</DocumentFragment>
`;

exports[`WelcomePage > renders the expected content in Spanish 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-omazcq-MuiStack-root"
  >
    <h1>
      Bienvenido a Dog Log!
    </h1>
    <p>
      Inicia sesi√≥n para continuar.
    </p>
    <button
      aria-label="Continue with Google"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
      data-testid="login-button"
      tabindex="0"
      type="button"
    >
      Continuar con Google
    </button>
  </div>
</DocumentFragment>
`;
